File: src/main/java/org/idempiere/common/base/IServicesHolder.java
package org.idempiere.common.base;

import java.util.List;

/**
 * 
 * @author hengsin
 *
 * @param <T>
 */
public interface IServicesHolder<T> {
	
	/**
	 * 
	 * @return list of service instance. null if not available or no matching service found
	 */
	public List<T> getServices();

}

File: src/main/kotlin/org/idempiere/common/base/IServicesHolder.kt
package org.idempiere.common.base

/**
 *
 * @author hengsin
 *
 * @param <T>
</T> */
interface IServicesHolder<T> {

    /**
     *
     * @return list of service instance. null if not available or no matching service found
     */
    val services: List<T>
}